
# Individual Project Planning Stage - Simran Heir



## Loading Datasets and Preforming Summary


In [1]:
# load tidyverse into the R notebook.
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors



### Player Dataset


In [2]:
# loading player data
url <- "https://raw.githubusercontent.com/SimranHeir/DSCI-100-Project-Planning-Stage-Individual-Simran-Heir/refs/heads/main/players.csv"
player_data <- read_csv (url)


Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [3]:
# printing out 6 rows of the player_data
head (player_data, 6)

experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

In [4]:
#How many observations are there for player data?
nrow (player_data)


[1] 196

In [5]:
#Player summary statistics
summary_player <- summary (player_data)
summary_player

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               


### Session Dataset


In [6]:
#loading session data
url <- "https://raw.githubusercontent.com/SimranHeir/DSCI-100-Project-Planning-Stage-Individual-Simran-Heir/refs/heads/main/sessions.csv"
sessiondata <- read_csv (url)


Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [7]:
# printing out 6 rows of the sessiondata
head (sessiondata, 6)

hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

In [8]:
#How many observations are there for session data?
nrow(sessiondata)

[1] 1535

In [9]:
#Session summary statistics
summary_session <- summary (sessiondata)
summary_session

 hashedEmail         start_time          end_time         original_start_time
 Length:1535        Length:1535        Length:1535        Min.   :1.712e+12  
 Class :character   Class :character   Class :character   1st Qu.:1.716e+12  
 Mode  :character   Mode  :character   Mode  :character   Median :1.719e+12  
                                                          Mean   :1.719e+12  
                                                          3rd Qu.:1.722e+12  
                                                          Max.   :1.727e+12  
                                                                             
 original_end_time  
 Min.   :1.712e+12  
 1st Qu.:1.716e+12  
 Median :1.719e+12  
 Mean   :1.719e+12  
 3rd Qu.:1.722e+12  
 Max.   :1.727e+12  
 NA's   :2          


# (1) Data Description:



## Where is data from?
The data is about information about players that play video games, the duration of playing video games, the player email, and whether they would subscribe to a game related newsletter.



## Player data:

- The data is about information about the players (age, gender, experience, and email), the time played, and whether they have subscribed.

- There are 7 variables: experience, subscribe, hashEmail, player_hours, name, gender, and age.

- There are 196 observations for this dataset.

|Variable|Type|Meaning|
|---------------|--------|--------|
|experience|factor|Experience level of the player|
|subscribe|logical|If a player is subscribed to the newletter or not (True or False)|
|hashEmail|character|An encoded email address to have privacy for the players|
|played_hours|double|Total amount of time that Minecraft was played (in hours)|
|name|character|Name of the players|
|gender|factor|Gender of the players|
|Age|integer|Age of the player (in years)|

For experience and gender they are incorrectly put as character, but they are factor for variable type. For Age it is incorrectly put as decimals, but is integer.


## Session data:

- The data is about the session. It includes the email, date and time played.

- There are 5 variables: hashEmail, start_time, end_time, original_start_time, and original_end_time.

- There are 1535 observations for this dataset.


|Variable|Type|Meaning|
|----|----|---|
|hashEmail|character|An encoded email adress to have privacy for the players|
|start_time|date + time|When the gameplay session started|
|end_time|date + time|When the gameplay session ended|
|original_start_time|double|Start time (in Unix Timestamp form)|
|original_end_time|double|End time (in Unix Timestamp form)|

For start_time and end_time it is originaly put as character, but it is a combination of date + time variable type.



# (2) Questions:



#### **Broad Question:** What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?

#### **Specific Question:** Can played_hours and experience predict subscription to a game related newsletter in a dataset?

For my questions it makes the most sense to use the player data since we are trying to determine what variables are most predictive of subscribing to a game related newsletter and only the player data set has the subscribe variable.

To wrangle the data first I will need to make sure each variable has the correct data type. 



## Wrangling Data - Player data



#### Give variables correct data type

In [10]:
# Give the variables the correct data type
player_data_wrangled <- player_data |>
   mutate(experience = as.factor (experience), gender = as.factor (gender), Age = as.integer (Age))


In [11]:
#print out the first 6 rows to see some of the dataset
head (player_data_wrangled, 6)

experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17


#### Calculate the mean for each value in player dataset


In [12]:
mean_player_data <- player_data_wrangled |>
    summarise(across(c (played_hours, Age),mean, na.rm = TRUE))
mean_player_data

Warning message:
“There was 1 warning in `summarise()`.
ℹ In argument: `across(c(played_hours, Age), mean, na.rm = TRUE)`.
Caused by warning:
! The `...` argument of `across()` is deprecated as of dplyr 1.1.0.
Supply arguments directly to `.fns` through an anonymous function instead.

  # Previously
  across(a:b, mean, na.rm = TRUE)

  # Now
  across(a:b, \(x) mean(x, na.rm = TRUE))”


played_hours Age     
1 5.845918     21.13918


## Activity #2: Explore Dataset Part 2 - Outcome Variable

Identify the main outcome/catergory/label variable?

How many values/groups are in this variable?

How many observations are there in each group?



## Activity #3: Explore Datasets Part 3 - Visualisations

Make some visualisations of the outcome variable:

what does the distribution of the variable look like

what relationship does it have with some of the other variables?


## (3) Exploratory Data Analysis and Visualization


## (4) Methods and Plan
